SG Branch

SaltyGraham opened the pull request:
Deleted BanishHackFix.java

Edited Randomtp.java (HIGHLY SUGGEST VIEWING PRIOR TO MERGING TO ENSURE THAT IT WORKS)

SaltyGraham committed:
Delete BanishHackFix.java

SaltyGraham committed:
Update RandomTP.java

I think this will automatically just RTP to the main world, unsure tho i suggest looking at it before merging

SaltyGraham committed:
Merge pull request #1 from SaltyGraham/patch-1

Delete BanishHackFix.java

jphuse commented on src/com/Jacksonnn/DCCore/RandomTP/RandomTP.java:
```diff
@@ -21,7 +21,7 @@ public boolean onCommand(CommandSender sender, Command command, String s, String
                     return true;
                 }
-                World world = player.getWorld();
+                World world = world;
```
Why did you set it to an uninitialized version of itself? That sets the value to itself, but since it doesn't have a value, to begin with, it's null.

We use player.getWorld() to mark the world they'll teleport in.

jphuse replied:
Is there a certain goal of editing RandomTP that you're trying to achieve?

SaltyGraham replied:
Essentially, I would like the RTP to be accessible anywhere and to be sent to the regular survival world.
The banish command was originally used to circumvent the dos player. This command along with /woosh, /piss and multiple others show up within the tab complete in the command line. 

Since the hacker has given up we have upgraded security, encryption, and we have our players protected by proxy. 

at this point having these commands accessible to regular players is a con that outways the pros.

jphuse reviewed (changes requested):
For the RandomTP, there are many bugs in the command, but setting `World world = world;` sets the variable to the uninitialized version of itself. This basically means that the variable `world` is set equal to itself, and if it doesn't have a value to begin with, it will end up being `null`.

------------

When deleting the BanishHackFix command, we need to remove all instances of it or the plugin won't compile:

- The CommandExecutor File (com.Jacksonnn.DCCore.StaffUtils.BanishHackFix.java - whole file)
- The Command Listener (com.Jacksonnn.DCCore.DCCore.java line 169)
- The plugin.yml (plugin.yml lines 110-112)

SaltyGraham committed:
Update plugin.yml

SaltyGraham committed:
Update DCCore.java

Removed Banish at 169

SaltyGraham committed:
Update RandomTP.java

Reverted Rtp

SaltyGraham commented:
Fixed Rtp

removed requested lines from dccore.java and plugin.yml

SaltyGraham committed:
Update plugin.yml

Instead of removing banish in plugin.yml, Commented out

SaltyGraham commented:
Restored banish in plugin.yml just commented out incase we need it in the future

SaltyGraham committed:
Update plugin.yml

Removed Banish from plugin.yml FINAL TIME lmao

SaltyGraham committed:
Update DCCore.java

Ok deleted banish for a final time lmao

SaltyGraham commented:
Ok I fully deleted Banish from Plugin.yml and Dccore.java

Celtican commented:
I still recommend inserting the @Deprecated annotation just before the class. It's a lot more clean than deleting the file, and it allows us to reuse the code should the need arise. Looks good though

SaltyGraham committed:
Update plugin.yml

Commented out /b override as we dont need it since we simply do not need voxelsniper

SaltyGraham committed:
Update PKAlias.java

Deprecated Pk Alias as we dont need voxelsniper

SaltyGraham committed:
Update DCCore.java

Commented out Pkalias for /b

SaltyGraham commented:
Deprecated and commented out Pkalias.java, 

We dont need voxelsniper and in tandem dont need to overwrite the command

Celtican commented:
/b removal looks much better 👍

SaltyGraham committed:
Update DCPlayerListener.java

What does line 53-63 do

SaltyGraham commented on src/com/Jacksonnn/DCCore/DCPlayerListener.java:
```diff
@@ -49,7 +49,7 @@ public void onLeave(PlayerQuitEvent e) {
     }
-
+ 
     @EventHandler
```
What does this look for on death

SaltyGraham replied:
@Jacksonnnnn What does this look for on death

jphuse replied:
@SaltyGraham It keeps the death and kill statistics, so on any death, it finds out if you were killed or the killer and increases the kill/death count.

SaltyGraham commented:
Jacksonnnnn can you view the changes please

Celtican reviewed (approved):
Looks good
